test/extensions/filters/listener/proxy_protocol/proxy_protocol_test.cc

```diff
@@ -210,6 +212,23 @@ INSTANTIATE_TEST_SUITE_P(IpVersions, ProxyProtocolTest,
                          testing::ValuesIn(TestEnvironment::getIpVersionsForTest()),
                          TestUtility::ipTestParamsToString);
 
+// This test ensures the socket file event was reset after timeout, otherwise
+// the assertion which avoid to create file event duplicated will be triggered.
+TEST_P(ProxyProtocolTest, Timeout) {
+  connect();
+  absl::SleepFor(absl::Seconds(5));
```

Comment:
Would it be possible to use simulated time instead of sleeping for such a long time?

Reply:
emm...this test is using a real dispatcher, so I can't mock it. I feel the whole proxy test is like an integration test, it listens and connects on real loopback.